xmtp/src/storage/encrypted_store/mod.rs

```diff
@@ -71,52 +60,60 @@ impl Default for EncryptedMessageStore {
 }
 
 impl EncryptedMessageStore {
-    pub fn new(
-        opts: StorageOption,
-        enc_key: EncryptionKey,
-    ) -> Result<Self, EncryptedMessageStoreError> {
+    pub fn new(opts: StorageOption, enc_key: EncryptionKey) -> Result<Self, StorageError> {
         let db_path = match opts {
             StorageOption::Ephemeral => ":memory:",
             StorageOption::Peristent(ref path) => path,
         };
 
-        let mut conn = SqliteConnection::establish(db_path)
-            .map_err(EncryptedMessageStoreError::DieselConnectError)?;
-
-        // Setup SqlCipherKey
-        Self::set_sqlcipher_key(&mut conn, &enc_key)?;
+        let pool = Pool::builder()
```

Comment:
I changed the mutex to a connection pool with a max size of 1.

They're functionally the same, but `r2d2` does all the heavy lifting of making this very friendly to be passed around. It comes with timeouts built in, can't be poisoned, stores on the heap, etc. We may have to switch back if we run into problems with the pragma encryption key getting reset on idle, but I haven't been able to reproduce any problems.